sdcm/nemesis.py

```diff
@@ -3533,33 +3530,32 @@ def decommission_nodes(self, add_nodes_number, rack, is_seed: Optional[Union[boo
                           f'{str(exc)}').publish()
 
     def disrupt_grow_shrink_cluster(self):
-        self._grow_cluster(rack=0)
-        self._shrink_cluster(rack=0)
+        self._grow_cluster(rack=None)
+        self._shrink_cluster(rack=None)
 
     def disrupt_grow_shrink_new_rack(self):
+        if not self._is_it_on_kubernetes():
+            raise UnsupportedNemesis("Adding new rack is not supported for non-k8s Scylla clusters")
         rack = max(self.cluster.racks) + 1
         self._grow_cluster(rack)
         self._shrink_cluster(rack)
 
-    def _grow_cluster(self, rack=0):
-        if rack > 0:
-            if not self._is_it_on_kubernetes():
-                raise UnsupportedNemesis("SCT rack functionality is implemented only on kubernetes")
-            self.log.info("Rack deletion is not supported on kubernetes yet. "
-                          "Please see https://github.com/scylladb/scylla-operator/issues/287")
-
+    def _grow_cluster(self, rack=None):
         add_nodes_number = self.tester.params.get('nemesis_add_node_cnt')
+        racks_count = len(self.tester.params.get("availability_zone").split(","))
```

Comment:
Current approach for creating Scylla racks on K8S backends in SCT doesn't depend on the AZs.
So, I expect it to have some problems on EKS backend where we specify always 2 AZs just to workaround EKS specific requirements, but use, de-facto, only first one.

Reply:
I don't get what is the problem. Replaced code:
```
        for _ in range(add_nodes_number):
            InfoEvent(message=f'GrowCluster - Add New node to {rack} rack').publish()
            added_node = self.add_new_node(rack=rack)
```
So rack was used in k8s tests. 
The only difference I see is that we grow cluster in all racks when on k8s (when rack is not specified as in `disrupt_grow_shrink_cluster`)

Reply:
We use single rack in K8S even when we specify 2 AZs in case of EKS.
The only situation when we create second rack it is using `disrupt_grow_shrink_new_rack` nemesis.

Reply:
ok, I changed the code so on k8s will use rack=0 when is not specified (`None`).